core/unit_test/tools/TestEventCorrectness.hpp

```diff
@@ -409,14 +409,19 @@ TEST(kokkosp, parallel_scan_no_fence) {
         << "skipping since the OpenMPTarget backend has unexpected fences";
 #endif
 
+  // Execute the parallel_scan first without looking for fence events.
+  // Depending on the backend implementation and the order of tests,
+  // it might be that the first call to parallel_scan is reallocating scratch
+  // memory which implies a fence when deallocating. We are not interested in
```

Comment:
Yes, the tests here are somewhat fragile. We want to test that certain API calls don't imply a fence but at the same time, we also prefer our internals to use the `Tools` interface so that our tools can provide more information.

The case we are running into here is one where we have to fence internally and the SYCL backend reports this fence. But in general, i.e., when run repeatedly, this API call should not imply a fence and that's how I adapted the test: from the second run on we should have gotten the scratch allocations right. Of course, this is somewhat backend-specific but in the end, almost all backends use some kind of scratch allocation for `parallel_scan`.